easybuild/easyconfigs/m/magma/magma-2.5.2-CrayGNU-19.10-cuda-10.1.eb

```diff
@@ -0,0 +1,48 @@
+# created by Anton Kozhevnikov (CSCS)
+
+easyblock = 'CMakeMake'
+
+name = 'magma'
+version = '2.5.2'
+versionsuffix = '-cuda-10.1'
+
+homepage = 'http://icl.cs.utk.edu/magma/'
+description = """The MAGMA project aims to develop a dense linear algebra
+library similar to LAPACK but for heterogeneous/hybrid architectures, starting
+with current Multicore+GPU systems."""
+
+toolchain = {'name': 'CrayGNU', 'version': '19.10'}
+toolchainopts = {'pic': True, 'openmp': True}
+
+sources = [SOURCE_TAR_GZ]
+source_urls = ['http://icl.cs.utk.edu/projectsfiles/magma/downloads/']
+
+builddependencies = [
+  ('CMake', '3.14.5', '', True),
+  ('cudatoolkit/10.1.105_3.27-7.0.1.1_4.1__ga311ce7', EXTERNAL_MODULE),
+  ('intel/19.0.1.144', EXTERNAL_MODULE)
+]
+
+dependencies = [
+  ('cudatoolkit/10.1.105_3.27-7.0.1.1_4.1__ga311ce7', EXTERNAL_MODULE),
+  ('intel/19.0.1.144', EXTERNAL_MODULE)
```

Comment:
Is this really using CrayGNU? It looks ambiguous by looking at the recipe: if users build software on top of this 'CrayGNU' module, they could end up using intel without noticing? 

you are often mixing both gcc and intel compilers in these recipes, maybe we should consider to create a specific toolchain?

Reply:
We want do compile with GNU, but link with MKL. MKL gives the best performance for linear algebra. For the second comment, we need to have a fresh gcc for Intel, otherwise it doesn't compile modern C++ code. So, we have to manually load gcc/8.2.0

Reply:
In this case, is it a problem then to add `-mkl` to the versionsuffix?

Can you rename the PR to match the version (2.6.2 -> 2.5.2)?

As for the need of gcc when using intel, I think that should be included in the toolchain (but we'll take that offline since this is unrelated to this PR).